pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -498,6 +503,38 @@ func (a *Actuator) ensureAnnotation(ctx context.Context, machine *machinev1beta1
 	return true, a.client.Update(ctx, machine)
 }
 
+func (a *Actuator) ensureProviderID(ctx context.Context, machine *machinev1beta1.Machine, host *bmh.BareMetalHost) (error) {
+	providerID := "baremetalhost:///" + host.Namespace + "/" + host.Name
+	existingProviderID := machine.Spec.ProviderID
+	if existingProviderID == nil || *existingProviderID != providerID {
+		log.Printf("Setting ProviderID %s for machine %s.", providerID, machine.Name)
+		machine.Spec.ProviderID = &providerID
+		err := a.client.Update(ctx, machine)
+		if err != nil {
+			log.Printf("Failed to update machine ProviderID, error: %s", err.Error())
+			return err
+		}
+	}
+
+	node, err := a.getNodeByMachine(ctx, machine)
+        if err != nil {
+                log.Printf("Failed to find node assosicated with Machine %s, error: %s", machine.Name, err.Error())
+//                return err
+		// FIXME should we ignore this?
+		return nil
+        }
+
+	log.Printf("Setting ProviderID %s for node %s.", providerID, node.Name)
+	node.Spec.ProviderID = providerID
+        err = a.client.Update(ctx, node)
```

Comment:
It looks like there's an indentation inconsistency here.

Reply:
Thanks yes, mix of tabs/spaces which should now be fixed